src/common/wallet-state.ts

```diff
@@ -210,6 +210,30 @@ const getStatus = (tx: AccountTransaction, currentBlock: number): Transaction['s
   else return 'confirmed'
 }
 
+export const mergeTransactions = (current: Transaction[]) => (
+  previous: Transaction[],
+): Transaction[] => {
+  const currentByHashes = new Map(current.map((tx) => [tx.hash, tx]))
+
+  const previousOutput = previous
```

Comment:
This should contain only failed transactions, right? Can't we find a better name than `previousOutput` then? :)

Reply:
Only failed and confirmed, etc. that are not present in the newer list